### PR TITLE
Update developer's guide and uml diagram

### PR DESCRIPTION
**Pull request by Chongjx**
#41 

Updated UML diagrams for Class structure, Parser, AddNoteCommand

Updated Tags section

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -81,19 +82,33 @@ NotUs manages the flow of the application. On launch, it will create the necessa
   <br><em>Figure #</em>
 </p>
 
-#### <a id="parser"><ins>2.3 Parser</ins></a>
+#### <a id="parserManager"><ins>2.3 ParserManager</ins></a>
 
-The Parser class handles all the user input from the command terminal. It makes sense of the user input and calls the respective commands into action.
+The ParserManager manages the creation of specific parser objects based on the type of the command. The parser then
+ makes sense of the user input and calls the respective commands into action.
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:88]:
```suggestion
The ParserManager manages the creation of specific parser objects based on the type of command. The parser then makes sense of the user input and calls the respective commands into action.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -81,19 +82,33 @@ NotUs manages the flow of the application. On launch, it will create the necessa
...
 
 1. Receives the user input message as a whole.
-1. Interprets the type of command and splits the message to identify all the parameters provided.
+1. Interprets the type of command and creates the respective parser.
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:91]:
```suggestion
1. Interprets the type of command and creates the respective parser for each command.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -58,7 +58,8 @@ Diagrams found in our documentation were generated using <a href="https://plantu
 The architecture design is given in the diagram above. The main components of NotUS are:
 
 1. `InterfaceManager`: Manages the user input as well as the message output from application.
-1. `Parser`: Makes sense of user input, and parses the information to the respective commands.
+1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The parser then
+ makes sense of the information to the respective commands.
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:62]:
```suggestion
1. `ParserManager`: Creates a suitable parser, based on the command, to make sense of user message. The respective parsers then make sense of the information and calls the respective commands.
```

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -81,19 +82,33 @@ NotUs manages the flow of the application. On launch, it will create the necessa
...
  
 <p align="center">
   <img alt="Parser" src="diagrams/out/Parser.png" />
   <br><em>Figure #</em>
 </p>
 
+💡 Note that the alternate paths in the sequence diagram above is not exhaustive. There is an alternate path for each
+ unique command. As there are too many paths, they are omitted from the diagram. The Command objects in the diagram is
+  used to represent a generic Command object that is created through the Parser. Refer to the next figure
+   for more details.
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:103]:
```suggestion
💡 Note that the alternate paths in the sequence diagram above are not exhaustive. There is an alternate path for each unique command. As there are too many paths, they are omitted from the diagram. The Command objects in the diagram are used to represent a generic Command object that is created through the Parser. Refer to the next figure for more details.
```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -102,6 +117,17 @@ The Parser class handles all the user input from the command terminal. It makes
...
+The class diagram below denotes the relationship between the TagManager and the Taggable Objects (Notes and Events).
+
+<p align="center">
+   <img alt="Parser" src="diagrams/out/TaggableObject.png"/>
+   <br><em>Figure #</em>
+ </p>
+ 
+Notes and Events inherit from the abstract class, TaggableObject and TagManager contains a map of individual unique
+ tags to an arraylist of TaggableObjects. The TagManager handles the creation, deletion as well as the tagging and
+  untagging of tags from notes or events.
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:129]:
```suggestion
Notes and Events inherit from the abstract class, TaggableObject, and TagManager contains a map of individual unique tags to an ArrayList of TaggableObjects. The TagManager handles the creation, deletion as well as the tagging and untagging of tags from notes or events.
```

File: docs/UserGuide.md [hunk 6]
```diff
@@ -297,43 +298,60 @@ Example of usage:
 
 Expected output:
 
+### <a id="tag-n"><ins>3.13 Tag/Untag Notes:</ins> `tag`</a>
```
Comment (R-Ramana) [inline docs/UserGuide.md:301]:
```suggestion
### <a id="tag-n"><ins>3.13 Tag/Untag Notes:</ins> `tag-n`</a>
```

File: docs/UserGuide.md [hunk 7]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
+Format: `tag-n /i INDEX /tag TAG [TAG COLOR] [/tag TAG]...`
 
-Format: `delete-t /tag TAG [/tag TAG]...`
+- Tags a note with a given TAG. If the note already contains the TAG, the tag will be removed from the note.The index refers to the index number shown in the displayed note list. The index must be a **positive integer** (1, 2, 3, …).
+- The user can create tag or untag multiple tags to a note within a single line.
 
-- Deletes the tag with the name TAG.
-- The user can delete multiple tags within a single line.
-- Notes with the tag, will have the tag removed.
+💡 If the user tries to tag a non existing tag to a note, the application will automatically create the tag for it.
```
Comment (R-Ramana) [inline docs/UserGuide.md:309]:
```suggestion
💡 If the user tries to tag a non-existing tag to a note, the application will automatically create a new tag, and add the tag to the note.
```

File: docs/UserGuide.md [hunk 8]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
 
 Example of usage: 
 
-`delete-t /tag CS2113`
+`Tag-n /i 1 /tag CS2113`
+
+`Tag-n /i 1 /tag CS2113 /tag important`
```
Comment (R-Ramana) [inline docs/UserGuide.md:315]:
```suggestion
`tag-n /i 1 /tag CS2113`

`tag-n /i 1 /tag CS2113 /tag important`
```

File: docs/UserGuide.md [hunk 9]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
 
 Expected output:
 
+### <a id="tag-e"><ins>3.14 Tag/Untag Events:</ins> `tag`</a>
```
Comment (R-Ramana) [inline docs/UserGuide.md:319]:
```suggestion
### <a id="tag-e"><ins>3.14 Tag/Untag Events:</ins> `tag-e`</a>
```

File: docs/UserGuide.md [hunk 10]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
-Format: `tag /i INDEX /tag TAG [TAG COLOR] [/tag TAG]...`
-
-- Tags a note with a given TAG. If the note already contains the TAG, the tag will be removed from the note.The index refers to the index number shown in the displayed note list. The index must be a **positive integer** (1, 2, 3, …).
-- The user can create tag or untag multiple tags to a note within a single line.
+- Tags a note with a given TAG. If the event already contains the TAG, the tag will be removed from the event.The index
+ refers to the index number shown in the displayed event list. The index must be a **positive integer** (1, 2, 3, …).
+- The user can create tag or untag multiple tags to an event within a single line.
 
-💡 If the user tries to tag a non existing tag to a note, the application will automatically create the tag for it.
+💡 If the user tries to tag a non existing tag to an event, the application will automatically create the tag for it.
```
Comment (R-Ramana) [inline docs/UserGuide.md:328]:
```suggestion
💡 If the user tries to tag a non-existing tag to a note, the application will automatically create a new tag, and add the tag to the note.
```

File: docs/UserGuide.md [hunk 11]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
 
 Example of usage: 
 
-`Tag /i 1 /tag CS2113`
+`Tag-e /i 1 /tag CS2113`
+
+`Tag-e /i 1 /tag CS2113 /tag important`
```
Comment (R-Ramana) [inline docs/UserGuide.md:334]:
```suggestion
`tag-e /i 1 /tag CS2113`

`tag-e /i 1 /tag CS2113 /tag important`
```

**Review by Nazryl — APPROVED** [review]

File: docs/UserGuide.md [hunk 12]
```diff
@@ -297,43 +298,60 @@ Example of usage:
...
+
+Expected output:
+
+### <a id="delete-t"><ins>3.15 Delete Tags:</ins> `delete-t`</a>
+Deletes a tag from the list of tags and remove the tag from the related notes.
```
Comment (R-Ramana) [inline docs/UserGuide.md:339]: Only tags? Can we do it for events as well?

**Review by R-Ramana — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Chongjx) [reply docs/DeveloperGuide.md:88]: This one is the ide auto ident to next line.

File: docs/UserGuide.md (hunk 10 above)
Comment (Chongjx) [reply docs/UserGuide.md:328]: This is for tag-e

File: docs/UserGuide.md (hunk 12 above)
Comment (Chongjx) [reply docs/UserGuide.md:339]: This is deleting of the tag and not untagging. Will update the content.

File: docs/UserGuide.md (hunk 10 above)
Comment (R-Ramana) [reply docs/UserGuide.md:328]: My apologies, I copy-pasted from the change I made for tag-n then forgot to update it.

**Review by R-Ramana — APPROVED** [review]

File: docs/UserGuide.md (hunk 10 above)
Comment (Chongjx) [reply docs/UserGuide.md:328]: Noted. Will edit it

**Pull request merged**